ddprof-lib/src/main/cpp/javaApi.cpp

```diff
@@ -379,23 +421,39 @@ Java_com_datadoghq_profiler_JavaProfiler_recordQueueEnd0(
 
 extern "C" DLLEXPORT jlong JNICALL
 Java_com_datadoghq_profiler_JavaProfiler_currentTicks0(JNIEnv *env,
-                                                       jobject unused) {
+                                                       jclass unused) {
+  return TSC::ticks();
+}
+
+extern "C" DLLEXPORT jlong JNICALL
+JavaCritical_com_datadoghq_profiler_JavaProfiler_currentTicks0() {
   return TSC::ticks();
 }
 
 extern "C" DLLEXPORT jlong JNICALL
 Java_com_datadoghq_profiler_JavaProfiler_tscFrequency0(JNIEnv *env,
-                                                       jobject unused) {
+                                                       jclass unused) {
+  return TSC::frequency();
+}
+
+extern "C" DLLEXPORT jlong JNICALL
+JavaCritical_com_datadoghq_profiler_JavaProfiler_tscFrequency0() {
```

Comment:
Similar note to `init0` -- this value seems to be saved in a `static final` on the Java side and never called again.